samples/Exporters/Console/TestApplicationInsights.cs

```diff
@@ -32,27 +32,25 @@ internal static object Run()
             DistributedContext.Carrier = AsyncLocalDistributedContextCarrier.Instance; // Enable asynclocal carrier for the context
             DistributedContext dc = DistributedContextBuilder.CreateContext(FrontendKey, "mobile-ios9.3.5");
 
-            using (var tracerFactory = TracerFactory.Create(builder => builder
+            using var tracerFactory = TracerFactory.Create(builder => builder
                 .SetResource(Resources.CreateServiceResource("my-service"))
-                .UseApplicationInsights(config => config.InstrumentationKey = "instrumentation-key")))
-            {
-                var tracer = tracerFactory.GetTracer("application-insights-test");
+                .UseApplicationInsights(config => config.InstrumentationKey = "instrumentation-key"));
+            var tracer = tracerFactory.GetTracer("application-insights-test");
 
-                using (DistributedContext.SetCurrent(dc))
```

Comment:
The problem with this is now `Dispose` will only be called at the end of the method thus changing the scope of the object.

That said, I would not apply this to the whole code base like that.

Reply:
I think you are wrong here. The new using syntax can increase the object lifecycle, but it's not happening here. `tracerFactory`, `DistributedContext.SetCurrent(dc)` and `tracer.StartActiveSpan` is disposed at the same time as they did before.

Comparison of the IL in Release: https://www.diffchecker.com/TMxLE1Si

Reply:
Does the diff show the original code? I can only see the IL.

I would expect it to change in this case since there are a few blocks one following the other.
So either I'm all the way wrong, meaning it doesn't increase the lifetime. In which case the compiler would need to inject the `Dispose` call right after the last use of the disposable (not my understanding of the feature, would need to check).

Reply:
OK now with split view it's easier to review.

This method could have:

```
               Thread.Sleep(TimeSpan.FromMilliseconds(5100));
            Thread.Sleep(TimeSpan.FromMilliseconds(5100));


                Console.WriteLine("Done... wait for events to arrive to backend!");
            Console.WriteLine("Done... wait for events to arrive to backend!");
                Console.ReadLine();
```
outside of the `using` block. Since it was inside, the lifetime of the first disposable is indeed the whole method so in this case it doesn't make a difference.

Instead of using ReSharper here, I'd reduce the scope of the disposable instead. (although not in this particular case because these are just tests so who cares).